opentelemetry-exporter-cloud-trace/tests/test_gcp_resource_detector.py

```diff
@@ -46,11 +57,75 @@ def test_finding_gce_resources(self, getter):
         )
 
 
+def pop_environ_key(key):
+    if key in os.environ:
+        os.environ.pop(key)
+
+
+def clear_gke_env_vars():
+    pop_environ_key("CONTAINER_NAME")
+    pop_environ_key("NAMESPACE")
+    pop_environ_key("HOSTNAME")
+
+
+class TestGKEResourceFinder(unittest.TestCase):
+    def tearDown(self) -> None:
+        clear_gke_env_vars()
+
+    @mock.patch("opentelemetry.tools.resource_detector.requests.get")
+    def test_missing_environment_variables(self, getter):
```

Comment:
Let's break this down into even smaller tests:

```
def test_missing_container_name(self, getter):
  getter.return_value.json.return_value = GKE_RESOURCES_JSON_STRING
  pop_environ_key("CONTAINER_NAME")
  self.assertEqual(get_gke_resources(), {})

def test_missing_namespace(self, getter):
  getter.return_value.json.return_value = GKE_RESOURCES_JSON_STRING
  pop_environ_key("NAMESPACE")
  self.assertEqual(get_gke_resources(), {})
```

(The case where both are missing might be overkill so we can skip that. Then these tests are clear enough that I don't think you need the comment in lines 79-80.)